### PR TITLE
clarify the `name` property needs to match between JS and PHP

### PR DESCRIPTION
**Pull request by netzkollektiv**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1.
2.
3.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [x] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 
This is a docs update.

</details>

### CODE REVIEW

**Comment by opr** [comment]
Hi - let's put this on hold as discussed in WooCommerce Community Slack - it seems incorrect that the `name` should have to match, whereas the `paymentMethodId` does not. We should investigate this and determine if it's something we should fix. If it's not something we will change, this PR looks good and we could merge it.

File: plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md [hunk 1]
```diff
@@ -205,7 +205,7 @@ First, you create a class that extends `Automattic\WooCommerce\Blocks\Payments\I
 
 In your class:
 
--   Define a `name` property (which is a string used to reference your payment method).
+-   Define a `name` property (it is important to use the same name as in your js payment registration, see above `my_payment_method`).
```
Comment (opr) [inline plugins/woocommerce-blocks/docs/third-party-developers/extensibility/checkout-payment-methods/payment-method-integration.md]:
```suggestion
-   Define a `name` property (This property is a string used to reference your payment method. It is important to use the same name as in your client-side JavaScript payment method registration, see `name: 'my_payment_method'` in the options object above).
```

**Review by opr — DISMISSED** [review]
Thanks for this PR @netzkollektiv - I wanted to add a little more context to the `name` property so I added it as a suggestion. If you agree please feel free to commit my suggestion. Thanks!

**Review by nielslange — COMMENTED** [review]
Thanks for working on this, @netzkollektiv. I wonder if you have seen @opr's suggestion in https://github.com/woocommerce/woocommerce/pull/48219#pullrequestreview-2104758811?

**Review by nielslange — APPROVED** [review]
Thanks for applying @opr's suggestion, @netzkollektiv. The change looks good to me. Thanks again for working on this.

@opr Do you want to take a final look at this PR?

**Review by opr — APPROVED** [review]
Thanks a lot @netzkollektiv we really appreciate you helping to improve the WooCommerce documentation! Hopefully we get the chance to work together again soon.

**Pull request merged**